### ISSUE TITLE
Adopt TimeProvider abstraction when its ready

### ISSUE THREAD

**Issue by stephentoub** [microsoft/semantic-kernel#609]
The DefaultHttpRetryHandler type defines IDelayProvider and ITimeProvider interfaces.  The APIs these surface are fully covered by the new TimeProvider abstract type being introduced in .NET 8 but that will also be available from the Microsoft.Bcl.TimeProvider nuget package downlevel.  Once it's available, it should ideally be used instead of Semantic Kernel defining its own abstractions.

**Comment by dluc** [comment]
Agreed, once we have a chance to revisit the unit tests approach, we'll start using this, removing the bespoke version
